File: solutions/2026-02-22/python/2026-02-22-binary-gap.py
```python
# Problem: Binary Gap
# Summary: Find the longest distance between adjacent 1s in the binary representation of a number.
# Link: https://leetcode.com/problems/binary-gap/

# Approach:
# 1. Convert the integer to its binary string representation.
# 2. Iterate through the binary string to find the positions of all '1's.
# 3. Keep track of the index of the last encountered '1'.
# 4. For each subsequent '1' found, calculate the distance from the previous '1' and update the maximum distance if the current distance is greater.
# 5. If no two '1's are found, the maximum distance remains 0.

# Time Complexity: O(log n) - The conversion to binary string takes O(log n) time, and iterating through the string also takes O(log n) time, where n is the input integer.
# Space Complexity: O(log n) - The binary string representation of n can have a length of O(log n).

class Solution:
    def binaryGap(self, n: int) -> int:
        # Convert the integer to its binary representation.
        # The bin() function returns a string like "0b10110".
        # We slice from index 2 to remove the "0b" prefix.
        binary_representation = bin(n)[2:]

        # Initialize the maximum gap found so far.
        max_gap = 0
        # Initialize the index of the last encountered '1'.
        # We use -1 to indicate that no '1' has been found yet.
        last_one_index = -1

        # Iterate through the binary string with both index and character.
        for i, bit in enumerate(binary_representation):
            # If the current bit is '1':
            if bit == '1':
                # Check if we have encountered a previous '1'.
                if last_one_index != -1:
                    # Calculate the distance between the current '1' and the last '1'.
                    # The distance is the difference in their indices.
                    current_gap = i - last_one_index
                    # Update the maximum gap if the current gap is larger.
                    max_gap = max(max_gap, current_gap)

                # Update the index of the last encountered '1' to the current index.
                last_one_index = i

        # Return the maximum gap found. If no adjacent 1s were found, max_gap will remain 0.
        return max_gap

```

File: solutions/2026-02-22/javascript/2026-02-22-binary-gap.js
// Problem Summary: Find the longest distance between adjacent 1s in a positive integer's binary representation.
// Link: https://leetcode.com/problems/binary-gap/
// Approach:
// 1. Convert the integer to its binary string representation.
// 2. Initialize `maxGap` to 0 and `lastOneIndex` to -1 (to indicate no '1' has been found yet).
// 3. Iterate through the binary string.
// 4. If a '1' is encountered:
//    a. If `lastOneIndex` is not -1 (meaning we've found a previous '1'), calculate the current gap (`currentIndex - lastOneIndex`).
//    b. Update `maxGap` if the current gap is larger.
//    c. Update `lastOneIndex` to the `currentIndex`.
// 5. Return `maxGap`.
// Time Complexity: O(log n) - The conversion to binary string takes logarithmic time with respect to n, and iterating through the string also takes time proportional to the number of bits, which is log n.
// Space Complexity: O(log n) - The space used to store the binary string.

/**
 * @param {number} n
 * @return {number}
 */
var binaryGap = function(n) {
    // Convert the integer to its binary string representation.
    const binaryString = n.toString(2);
    
    // Initialize maxGap to 0. This will store the longest distance found.
    let maxGap = 0;
    
    // Initialize lastOneIndex to -1. This will store the index of the most recently encountered '1'.
    // A value of -1 indicates that no '1' has been seen yet.
    let lastOneIndex = -1;

    // Iterate through each character (bit) of the binary string.
    for (let i = 0; i < binaryString.length; i++) {
        // Check if the current character is '1'.
        if (binaryString[i] === '1') {
            // If a '1' is found and we've already seen a previous '1' (lastOneIndex is not -1).
            if (lastOneIndex !== -1) {
                // Calculate the distance between the current '1' and the last '1'.
                const currentGap = i - lastOneIndex;
                // Update maxGap if the current gap is larger than the maximum gap found so far.
                maxGap = Math.max(maxGap, currentGap);
            }
            // Update lastOneIndex to the current index 'i', as this is now the most recent '1'.
            lastOneIndex = i;
        }
    }

    // Return the maximum gap found. If no two adjacent 1s were found, maxGap will remain 0.
    return maxGap;
};
```